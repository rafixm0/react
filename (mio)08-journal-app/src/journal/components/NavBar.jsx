import { AppBar, Grid, IconButton, Toolbar, Typography } from "@mui/material"
import { LogoutOutlined, MenuBookOutlined } from "@mui/icons-material"


export const NavBar = ({ drawerWidth = 240 }) => {
  return (
    <AppBar 
        position="fixed"
        sx={{ 
            width: {sm: `calc(100% - ${ drawerWidth }px)` },
            ml: { sm: `${ drawerWidth }` },
        }}
    >
        <Toolbar>
            <IconButton
                color='inherit'
                edge="start"
                sx={{ mr: 2, display: { sm: 'none'} }}
            >
                <MenuBookOutlined />
            </IconButton>

            <Grid
                container
                direction='row'
                justifyContent='space-around'
                alignItems='center'
            >
                <Typography variant="h6" noWrap component='div'>JournalApp</Typography>
                <IconButton color="error">
                    <LogoutOutlined />
                </IconButton>

            </Grid>

        </Toolbar>
    </AppBar>
  )
}
