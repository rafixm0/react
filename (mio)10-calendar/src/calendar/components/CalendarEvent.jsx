

export const CalendarEvent = ({ event }) => {

    const { title , user } = event;
    
    console.log(event);
    console.log(title);
    console.log(user.name);
  
    return (
    <>
        <strong>{ title }</strong>
        <span> - { user.name }</span>
    </>
  )
}
