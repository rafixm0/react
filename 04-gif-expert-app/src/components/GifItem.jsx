

export const GifItem = ({ title, url, id }) => {
  
    //console.log({ title, url, id});

    return (
        <div className='card'>
            <img src={url} alt={title} />
        </div>
    )

}
