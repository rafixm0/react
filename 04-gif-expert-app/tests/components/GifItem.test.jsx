import { render } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('Pruebas en <GifItem />', () => {

    const title = 'GifExpertApp';
    const url = 'https://rafix.org';


    test('Debe de hacer match con el snapshoot', () => {

        const {container} = render ( <GifItem title={ title } url={ url } />);
        expect( container ).toMatchSnapshot();

    })


})