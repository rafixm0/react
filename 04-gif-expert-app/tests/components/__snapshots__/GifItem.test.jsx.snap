// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> Debe de hacer match con el snapshoot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="GifExpertApp"
      src="https://rafix.org"
    />
  </div>
</div>
`;
